chopper/lib/src/interceptors/http_logging_interceptor.dart

```diff
@@ -75,116 +76,152 @@ enum Level {
+    final Request request = chain.request;
+
+    final Stopwatch stopWatch = Stopwatch()..start();
+
+    final Response<BodyType> response = await chain.proceed(request);
+
+    stopWatch.stop();
+
+    if (level == Level.none || (onlyErrors && response.statusCode < 400)) {
+      return response;
+    }
+
     final http.BaseRequest baseRequest = await request.toBaseRequest();
 
-    String startRequestMessage =
-        '--> ${baseRequest.method} ${baseRequest.url.toString()}';
-    String bodyRequestMessage = '';
+    final StringBuffer startRequestMessage = StringBuffer(
+      '--> ${baseRequest.method} ${baseRequest.url.toString()}',
+    );
+    final StringBuffer bodyRequestMessage = StringBuffer();
     if (baseRequest is http.Request) {
       if (baseRequest.body.isNotEmpty) {
-        bodyRequestMessage = baseRequest.body;
+        bodyRequestMessage.write(baseRequest.body);
 
         if (!_logHeaders) {
-          startRequestMessage += ' (${baseRequest.bodyBytes.length}-byte body)';
+          startRequestMessage.write(
+            ' (${baseRequest.bodyBytes.length}-byte body)',
+          );
         }
       }
     }
 
     // Always start on a new line
     _logger.info(ChopperLogRecord('', request: request));
-    _logger.info(ChopperLogRecord(startRequestMessage, request: request));
+    _logger.info(
+      ChopperLogRecord(startRequestMessage.toString(), request: request),
+    );
 
     if (_logHeaders) {
       baseRequest.headers.forEach(
-        (k, v) => _logger.info(ChopperLogRecord('$k: $v', request: request)),
+        (String k, String v) => _logger.info(
+          ChopperLogRecord('$k: $v', request: request),
+        ),
       );
 
       if (baseRequest.contentLength != null &&
           baseRequest.headers['content-length'] == null) {
-        _logger.info(ChopperLogRecord(
-          'content-length: ${baseRequest.contentLength}',
-          request: request,
-        ));
+        _logger.info(
+          ChopperLogRecord(
+            'content-length: ${baseRequest.contentLength}',
+            request: request,
+          ),
+        );
       }
     }
 
     if (_logBody && bodyRequestMessage.isNotEmpty) {
       _logger.info(ChopperLogRecord('', request: request));
-      _logger.info(ChopperLogRecord(bodyRequestMessage, request: request));
+      _logger.info(
+        ChopperLogRecord(bodyRequestMessage.toString(), request: request),
+      );
     }
 
     if (_logHeaders || _logBody) {
-      _logger.info(ChopperLogRecord(
-        '--> END ${baseRequest.method}',
-        request: request,
-      ));
+      _logger.info(
+        ChopperLogRecord('--> END ${baseRequest.method}', request: request),
+      );
     }
-    final stopWatch = Stopwatch()..start();
-
-    final response = await chain.proceed(request);
 
-    stopWatch.stop();
+    if (level == Level.none || (onlyErrors && response.statusCode < 400)) {
+      return response;
+    }
```

Comment:
Duplicate code? Its also at the beginning of the method

Reply:
Yeah, it was there in the original file https://github.com/lejard-h/chopper/blob/707950bbb3379ea2332b040815b9205c3184c40e/chopper/lib/src/interceptors/http_logging_interceptor.dart#L143 so I just added the additional condition. But on second thought, I should remove it.